frontend/src/components/SliceDropdown.vue

```diff
@@ -0,0 +1,189 @@
+              <v-list-item-icon>
+                <v-icon>edit</v-icon>
+              </v-list-item-icon>
+            </v-list-item>
+            <v-list-item @click.stop="deleteSlice(data.item)">
+              <v-list-item-title>Delete</v-list-item-title>
+              <v-list-item-icon>
+                <v-icon>delete</v-icon>
+              </v-list-item-icon>
+            </v-list-item>
+          </v-list>
+        </v-menu>
+      </template>
+    </v-autocomplete>
+
+    <v-dialog v-model="sliceDialog" width="800" persistent>
+      <v-card>
+        <ValidationObserver ref="dialogForm">
+          <v-form @submit.prevent="submit()">
+            <v-card-title v-if="isUpdateDialog">Edit slice {{ sliceName }}</v-card-title>
+            <v-card-title v-else>Create new slice</v-card-title>
+            <v-card-text>
+              <ValidationProvider name="Name" mode="eager" rules="required" v-slot="{errors}">
+                <v-text-field label="Slice Name*" type="text" v-model="sliceName"
+                              :error-messages="errors"></v-text-field>
+              </ValidationProvider>
+              <span>Python function</span>
+              <MonacoEditor
+                  ref="editor"
+                  v-model='sliceCode'
+                  class='editor'
+                  language='python'
+                  style="height: 300px"
+                  :options="$root.monacoOptions"
+              />
+              <!--              <div v-if="projectCreateError" class="caption error&#45;&#45;text">{{ projectCreateError }}</div>-->
+            </v-card-text>
+            <v-card-actions>
+              <v-spacer></v-spacer>
+              <v-btn color="error" text @click="closeSliceDialog()">Cancel</v-btn>
+              <v-btn color="primary" text type="submit">
+                <div v-if="isUpdateDialog">Update</div>
+                <div v-else>Create</div>
+              </v-btn>
+            </v-card-actions>
+          </v-form>
+        </ValidationObserver>
+      </v-card>
+    </v-dialog>
+  </div>
+</template>
+
+<script setup lang="ts">
+// @ts-ignore
+import MonacoEditor from 'vue-monaco';
+import {computed, onMounted, ref, watch} from "vue";
+import {SliceDTO} from "@/generated-sources";
+import {api} from "@/api";
+
+interface Props {
+  projectId: number,
+  loading?: boolean,
+  create?: boolean,
+  isProjectOwnerOrAdmin?: boolean
+}
+
+const props = defineProps<Props>();
+const emit = defineEmits(["onSelect", "onClear"])
+
+const slices = ref<SliceDTO[]>([])
+const slice = ref<SliceDTO | null>(null);
+
+const sliceDialog = ref<boolean>(false);
+const sliceName = ref<string>("");
+const sliceCode = ref<string>("");
+const sliceId = ref<number>(-1);
+
+const autocomplete = ref<any | null>(null);
+
+onMounted(() => {
+  loadSlices();
+})
+
+const items = computed(() => {
+  // TODO: Find a solution for this typing issue
+  let result: any[] = slices.value.map(slice => ({
```

Comment:
Maybe have a type `AutocompleteItem` defined as following

```
{
  text: string,
  name: string,
  index: number | null
}
```

For each item in the `slices` array have either the index in the array or ID of the slice to retrieve it latter on and for the `__NEW_SLICE__` item have `-1`

I think it would be cleaner.